src/fondue/Fondue.js

```diff
@@ -760,7 +777,45 @@ export default class Fondue {
 				});
 			}
 
-			return currentAllGlyphs;
+			// Chained context substitution
+			if (lookup.lookupType === 6) {
+				lookup.subtableOffsets.forEach((_, i) => {
+					let subtable = lookup.getSubTable(i);
+
+					if (subtable.inputGlyphCount > 0) {
+						subtable.inputCoverageOffsets.forEach((offset) => {
+							const coverage = subtable.getCoverageFromOffset(
+								offset
+							);
+							parsedLookup["input"][i] = glyphToLetter(coverage);
```

Comment:
Since you're assigning based on index (`i`) here and these assignments are wrapped in conditions, it's possible that `parsedLookup["input"]` contains 'gaps`, i.e. an array with `[undefined, 'A', undefined, undefined, 'B']` etc. Is that intentional and something that can actually happen? It feels to me like this might be quite surprising as a consumer of this function.

Reply:
`glyphToLetter` will actually return an array of characters for an input/backtrack/lookahead.

These are empty by default (done [here](https://github.com/Wakamai-Fondue/wakamai-fondue-engine/blob/c3bd44a214157e42ae1183bd2144d6040fedd279/src/fondue/Fondue.js#L685)) and will only be filled when this condition is true. Either outcome is fine and expected.

So this is working differently than you think, I think :-) Does that make sense?

Reply:
Right, I see. `glyphToLetter` is perhaps a bit misleading name then?

Reply:
Could you take a look at the refactored code to see if it's still confusing?

Reply:
Makes sense now! :+1: